### PR TITLE
Optifine D1 fix

### PR DESCRIPTION
**Pull request by Foghrye4**
Fixes compatibility with Optifine D1 (and probably higher).

### CODE REVIEW

File: src/main/java/cubicchunks/asm/CubicChunksMixinConfig.java [hunk 1]
```diff
@@ -52,9 +57,50 @@
...
     @Override
     public void onLoad(String mixinPackage) {
+        OptifineState optifineState = OptifineState.NOT_LOADED;
+        try {
+            Class optifineInstallerClass = Class.forName("optifine.Installer");
+            Method getVersionHandler = optifineInstallerClass.getMethod("getOptiFineVersion", new Class[0]);
+            String optifineVersion = (String) getVersionHandler.invoke(null, new Object[0]);
+            optifineVersion = optifineVersion.replace("_pre", "");
+            optifineVersion = optifineVersion.substring(optifineVersion.length() - 2, optifineVersion.length());
+            if (optifineVersion.compareTo("D1") >= 0)
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/CubicChunksMixinConfig.java]: That definitely doesn't look like reliable version detection. Aside of making or more reliable, an option to override it with `-Dcubicchunks.somethhing=something` would be nice.

File: src/main/java/cubicchunks/asm/CubicChunksMixinConfig.java [hunk 2]
```diff
@@ -52,9 +57,50 @@
...
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
+        } catch (SecurityException e) {
+            e.printStackTrace();
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
+        } catch (InvocationTargetException e) {
+            e.printStackTrace();
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/CubicChunksMixinConfig.java]: Handle them properly.

Comment (Foghrye4) [reply src/main/java/cubicchunks/asm/CubicChunksMixinConfig.java]: Re-throw something like `new Exception("Incompatible Optifine version. Use C7 or D1")`?

File: src/main/java/cubicchunks/asm/CubicChunksMixinConfig.java (hunk 1 above)
Comment (Foghrye4) [reply src/main/java/cubicchunks/asm/CubicChunksMixinConfig.java]: Optifine itself use a really hackish way to do a same thing. It seek for certain byte pattern in "InstallerFrame.class" bytecode and read bytes containing string field of an Optifine version. I have no idea for any more reliable method.

**Comment by Foghrye4** [comment]
Done.

**Pull request merged**